packages/flutter/lib/src/widgets/system_context_menu.dart

```diff
@@ -119,6 +159,258 @@ class _SystemContextMenuState extends State<SystemContextMenu> {
+  const IOSSystemContextMenuItemCut();
+
+  @override
+  IOSSystemContextMenuItemDataCut _getData(WidgetsLocalizations? localizations) {
+    return const IOSSystemContextMenuItemDataCut();
+  }
+}
+
+/// Creates an instance of [IOSSystemContextMenuItem] for the system's built-in
+/// paste button.
+///
+/// Should only appear when the field can receive pasted content.
+///
+/// The title and action are both handled by the platform.
+///
+/// See also:
+///
+///  * [SystemContextMenu], a widget that can be used to display the system
+///    context menu.
+///  * [IOSSystemContextMenuItemPaste], which specifies the data to be sent
+///     to the platform for this same button.
+class IOSSystemContextMenuItemPaste extends IOSSystemContextMenuItem {
+  /// Creates an instance of [IOSSystemContextMenuItemPaste].
+  const IOSSystemContextMenuItemPaste();
+
+  @override
+  IOSSystemContextMenuItemDataPaste _getData(WidgetsLocalizations? localizations) {
+    return const IOSSystemContextMenuItemDataPaste();
+  }
+}
+
+/// Creates an instance of [IOSSystemContextMenuItem] for the system's built-in
+/// select all button.
+///
+/// Should only appear when the field can have its selection changed.
+///
+/// The title and action are both handled by the platform.
+///
+/// See also:
+///
+///  * [SystemContextMenu], a widget that can be used to display the system
+///    context menu.
+///  * [IOSSystemContextMenuItemSelectAll], which specifies the data to be
+///     sent to the platform for this same button.
+class IOSSystemContextMenuItemSelectAll extends IOSSystemContextMenuItem {
+  /// Creates an instance of [IOSSystemContextMenuItemSelectAll].
+  const IOSSystemContextMenuItemSelectAll();
+
+  @override
+  IOSSystemContextMenuItemDataSelectAll _getData(WidgetsLocalizations? localizations) {
+    return const IOSSystemContextMenuItemDataSelectAll();
+  }
+}
+
+/// Creates an instance of [IOSSystemContextMenuItem] for the
+/// system's built-in look up button.
+///
+/// Should only appear when content is selected.
+///
+/// The [title] is optional, but it must be specified before being sent to the
+/// platform. Typically it should be set to
+/// [WidgetsLocalizations.lookUpButtonLabel].
+///
+/// The action is handled by the platform.
+///
+/// See also:
+///
+///  * [SystemContextMenu], a widget that can be used to display the system
+///    context menu.
+///  * [IOSSystemContextMenuItemLookUp], which specifies the data to be sent
+///    to the platform for this same button.
+class IOSSystemContextMenuItemLookUp extends IOSSystemContextMenuItem {
+  /// Creates an instance of [IOSSystemContextMenuItemLookUp].
+  const IOSSystemContextMenuItemLookUp({this.title});
+
+  @override
+  final String? title;
+
+  @override
+  IOSSystemContextMenuItemDataLookUp _getData(WidgetsLocalizations? localizations) {
+    return IOSSystemContextMenuItemDataLookUp(title: title ?? localizations!.lookUpButtonLabel);
```

Comment:
I wonder if the `localizations` parameter here should just be `required`. It's not a big concern though since this method is private.

Reply:
Yeah I think you're right, I'll make that change. I think it's simpler if it's required.